spring-cloud-gcp-storage/src/main/java/com/google/cloud/spring/storage/GoogleStorageResource.java

```diff
@@ -113,14 +114,13 @@ public boolean isAutoCreateFiles() {
 		return this.autoCreateFiles;
 	}
 
+	/**
+	 * @throws StorageException if an issue occurs getting the Bucket or Blob.
+	 * @return Returns true if the bucket or object exists.
+	 */
 	@Override
 	public boolean exists() {
-		try {
-			return isBucket() ? getBucket() != null : getBlob() != null;
-		}
-		catch (IOException ex) {
-			return false;
```

Comment:
This might be a breaking change. I imagine that previously, if there was a permissions issue, it would return `false`, but now it will throw an exception.

Reply:
Tested both with and without the change using a service account without permission, and both ended up throwing a `StorageException` that is not caught (because `StorageException` does not extend `IOException`), so _that_ behavior is the same.

Exception from the logs:
```
Request processing failed; nested exception is com.google.cloud.storage.StorageException: 897650840564-jriaec6tjb5n9od3fpgu63lgib25kqhk@developer.gserviceaccount.com does not have storage.objects.get access to the Google Cloud Storage object.
```

I'll add some docs.